apps/ensindexer/ponder.config.ts

```diff
@@ -68,51 +35,14 @@ const ponderConfig = activePlugins
 
 // set the indexing behavior dependencies
 ponderConfig.indexingBehaviorDependencies = {
-  HEAL_REVERSE_ADDRESSES: healReverseAddresses(),
+  HEAL_REVERSE_ADDRESSES: config.healReverseAddresses,
 };
 
-////////
-// Invariant: if using a custom START_BLOCK or END_BLOCK, ponder should be configured to index at
-// most one network.
-////////
-
-const globalBlockrange = getGlobalBlockrange();
-if (globalBlockrange.startBlock !== undefined || globalBlockrange.endBlock !== undefined) {
-  const numNetworks = Object.keys(ponderConfig.networks).length;
-  if (numNetworks > 1) {
-    throw new Error(
-      `ENSIndexer's behavior when indexing _multiple networks_ with a _specific blockrange_ is considered undefined (for now). If you're using this feature, you're likely interested in snapshotting at a specific END_BLOCK, and may have unintentially activated plugins that source events from multiple chains.
-
-The config currently is:
-ENS_DEPLOYMENT_CHAIN=${getEnsDeploymentChain()}
-ACTIVE_PLUGINS=${requestedPluginNames.join(",")}
-START_BLOCK=${globalBlockrange.startBlock || "n/a"}
-END_BLOCK=${globalBlockrange.endBlock || "n/a"}
-
-The usage you're most likely interested in is:
-  ENS_DEPLOYMENT_CHAIN=(mainnet|sepolia|holesky) ACTIVE_PLUGINS=subgraph END_BLOCK=x pnpm run start
-which runs just the 'subgraph' plugin with a specific end block, suitable for snapshotting ENSNode and comparing to Subgraph snapshots.
-
-In the future, indexing multiple networks with network-specific blockrange constraints may be possible.`,
-    );
-  }
-}
-
-////////
-// Invariant: All configured networks must have a custom RPC endpoint provided. Public RPC endpoints
-// will ratelimit and make indexing more or less unusable.
-////////
-
-// if not every network has a valid transport, panic
 const allChainIds = Object.values(ponderConfig.networks).map((network) => network.chainId);
-if (!allChainIds.every((chainId) => rpcEndpointUrl(chainId) !== undefined)) {
-  throw new Error(`ENSNode has been configured with the following ACTIVE_PLUGINS: ${requestedPluginNames.join(", ")}.
-These plugins, collectively, index events from the following chains: ${allChainIds.join(", ")}.
 
-The following RPC_URL_* environment variables must be defined for nominal indexing behavior:
-${allChainIds.map((chainId) => `RPC_URL_${chainId}: ${rpcEndpointUrl(chainId) || "N/A"}`).join("\n")}
-`);
-}
+// Validate the config before we activate the plugins. These validations go beyond simple type
+// validations and ensure any relationships between environment variables are correct.
+validateConfig(allChainIds, ponderConfig.networks);
```

Comment:
this would read better as `validatePonderConfig(config, ponderConfig)` where config is the zod-parsed `Config` and `ponderConfig` is the runtime-constructed ponder config